src/python/pants/backend/jvm/tasks/jvm_compile/jvm_compile.py

```diff
@@ -582,8 +586,38 @@ def execute(self):
                             classpath_product.remove_for_target(cc.target, [(conf, cc.classes_dir)])
                             classpath_product.add_for_target(cc.target, [(conf, cc.jar_file)])
 
-        if dependees_of_target_roots is not None:
-            self.create_extra_products_for_targets(dependees_of_target_roots)
+        if modulizable_targets is not None:
+            compilable_modulizable_targets = set(
+                filter(lambda x: self.select(x), modulizable_targets)
+            )
+            self.create_extra_products_for_targets(compilable_modulizable_targets)
+
+    def calculate_jvm_modulizable_targets(self):
+        def is_jvm_or_resource_target(t):
+            return isinstance(t, (JvmTarget, JvmApp, JarLibrary, Resources))
+
+        jvm_and_resources_target_roots = set(
+            filter(is_jvm_or_resource_target, self.context.target_roots)
+        )
+        jvm_and_resources_target_roots_minus_thrift_addresses = set(
```

Comment:
Nit: If we ever introduce protobuf, this would be a lie. Could you call this variable `jvm_and_resources_target_roots_minus_synthetic_addresses`?

Reply:
adjusted.